[test] add types to composite_compliance.py

coconutruben opened the pull request:
Summary:
# Why

Adding types to help static type checkers and runtime find issues proactively

# What

- annotate obvious types across the file
- use `Any` liberally when types are hard to annotate e.g. when a class is dynamically generated

Test Plan:
```
buck test //caffe2/caffe2:caffe2_test_gpu
```

Differential Revision: D63149677

henrylhtsang reviewed (approved):
LGTM. Merge after ci is done.

nit: fix title of this PR / diff

coconutruben committed:
[test] add types to composite_compliance.py (#136385)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/136385

# Why

Adding types to help static type checkers and runtime find issues proactively

# What

- annotate obvious types across the file
- use `Any` liberally when types are hard to annotate e.g. when a class is dynamically generated

Test Plan:
```
buck test //caffe2/caffe2:caffe2_test_gpu
```

Reviewed By: henrylhtsang

Differential Revision: D63149677
